ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -1612,6 +1612,17 @@ func (r *RayClusterReconciler) updateRayClusterStatus(ctx context.Context, origi
 	if err != nil {
 		logger.Info("Error updating status", "name", originalRayClusterInstance.Name, "error", err, "RayCluster", newInstance)
 	}
+
+	// Record the ray_cluster_provisioned_duration_seconds metric only after a successful status update
```

Comment:
Could you create a separate utility function `collectRayClusterMetrics(collector, newClusterStatus, oldClusterStatus, creationTimeStamp)` for this? Ideally, all logic for recording RayCluster metrics should be encapsulated within that function.

Adding a util function also allows us easier to write unit tests.